### ISSUE TITLE
Weird problem loading `Units\NightElf\Chimaera\Chimaera.mdx` model

### ISSUE THREAD

**Issue by jp06** [flowtsohg/mdx-m3-viewer#62]
I just noticed that this model only loads at certain circumstances. When it doesn't work, it just flashes the model's white silhouette for a split second or sometimes nothing at all (and there's no error on the console). It only successfully loads when it's the first model loaded, or when you load certain models first in case you loaded something else (I happened to notice this with the `Units\Demon\Kiljaeden\Kiljaeden.mdx` model before the `Chimaera.mdx`). There may be other models that have the same behavior that I am not aware of. As far as I'm aware, my code's just fine, I never noticed the same problem with the other hundred or so models I'm loading.

I'm using Preact and here's the code for my hero model viewer component, in case I'm doing something really bad here:
```javascript
import { h } from 'preact';
import { useRef, useEffect } from 'preact/hooks';

import ModelViewer from 'mdx-m3-viewer';

import EditBoxBackdrop from "../edit-box-backdrop"
import setupCamera from '../../utilities/camera'

import './hero-model-viewer.scss';

const HeroModelViewer = ({ model, mapVersion, scale, modelScale, rgb }) => {
	const canvasRef = useRef(null)
	let viewer

	useEffect(() => {
		viewer = viewer ? viewer : new ModelViewer.ModelViewer(canvasRef.current)
		canvasRef.current.height = 250

		const scene = viewer.addScene()
		scene.color = [0.02, 0.06, 0.02]
		// scene.color = [0.04, 0, 0.04]

		setupCamera(scene, {
			horizontalAngle: Math.PI / 1.5,
			verticalAngle: Math.PI / 2,
			distance: 350,
			lockVerticalRotate: true
		})
		
		function pathSolver(src) {
			const resourceServer = "http://localhost:4000"
			const resource = (src.includes(".") ? src : src + ".mdx").replace("\\", "/")
			
			return `${resourceServer}/${resource}?map=${mapVersion}`
		}
		
		viewer.addHandler(ModelViewer.handlers.mdx, pathSolver)
		viewer.addHandler(ModelViewer.handlers.blp)
		viewer.load(model, pathSolver).then(model => {
			if (model) {
				const instance = model.addInstance();
				const firstStandSequence = model.sequences.findIndex(({ name }, i) => name.toLowerCase().includes("stand"))

				instance.setScene(scene);
				instance.setSequence(firstStandSequence); 
				instance.setTeamColor(1);
				instance.setSequenceLoopMode(2);
				instance.uniformScale(scale)
				instance.setUniformScale(modelScale)
				instance.setLocation([0, 0, -75]);
				instance.setVertexColor(rgb.map(c => c / 255))
			}
		});

		(function step() {
			requestAnimationFrame(step);
			if (canvasRef.current?.parentNode) {
				let rect = canvasRef.current.parentNode.getBoundingClientRect()
				canvasRef.current.width = rect.width
			}
			viewer.updateAndRender();
		})();

		return () => viewer.clear()
	}, [model, mapVersion])

	return (
		<div class="hero-model-viewer">
			<EditBoxBackdrop>
				<div class="canvas-container">
					<canvas ref={canvasRef} />
				</div>
			</EditBoxBackdrop>
		</div>
	);
};

export default HeroModelViewer;
```

**Comment by flowtsohg** [comment]
Hmmm, I can't replicate this behavior.

Are you listening to viewer errors? `viewer.on('error', console.log)`

**Comment by jp06** [comment]
I actually did that too but I removed it when I saw that it doesn't emit any error or warning on that model. I think I'll come back to this issue later when I upload my code to GitHub after adding some things and doing some cleanups so others can reproduce.

**Comment by jp06** [comment]
This certain issue seems to be fixed along when I rewrote my viewer code when I'm fixing something else and adding some stuff. I misunderstood how my code works previously. It seems that with the above code, it will still refresh the variables every time it re-renders. I stored the values of `viewer`, `scene`, and model `instance`s using the `useRef` hook for it to persist instead of a local variable. For now, it seems to be switching between camera views, sequences and models seamlessly. Here's a preview (a bit sloppy of a recording and an over-clicking mouse sorry):

https://user-images.githubusercontent.com/30029476/108014128-35f2e880-7048-11eb-83a8-7a272d4d3bbb.mp4
